headers.c

```diff
@@ -430,7 +430,11 @@ GC_prev_block(struct hblk *h)
         j -= (GC_signed_word)ADDR(hhdr);
       } else {
         /* TODO: return hhdr -> hb_block instead */
+#ifndef CHERI_PURECAP
```

Comment:
Unfortunately this helps only with small tests. E.g. gctest fails at some moment because hhdr->hb_block is null (but it should not be null). A mentioned in the TODO above, we should return hb_block for all platforms but to do this we need to fix setting hb_block in allchblk.c. (I'm struggling with it these days but have not found a solution yet.)